### PR TITLE
Statviz frontend refactoring

### PR DESCRIPTION
**Pull request by MaikNeubert**

### CODE REVIEW

File: statviz/src/components/ErrorCard.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Box } from "@chakra-ui/react";
+
+export default function ErrorCard(props: { error: string }) {
+  return (
+    <Box>
```
Comment (HaGuesto) [inline statviz/src/components/ErrorCard.tsx:5]: This should be an alert

File: statviz/src/utils/theme.ts [hunk 2]
```diff
@@ -213,3 +214,230 @@ export const nivoScheme: Theme = {
...
+      chip: {},
+      table: {},
+      tableCell: {},
+      tableCellValue: {},
+    },
+  };
+};
+
+// Calculate margins, fontSizes, and styles for additional information that can be displayed in an exported image
+export const graphMarginTopPercent = 10;
```
Comment (HaGuesto) [inline statviz/src/utils/theme.ts:342]: I would split this up in nivo theme, visx theme, chakra theme and maybe styling helpers for exporting

File: statviz/src/views/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -21,11 +12,7 @@ export default function Dashboard() {
 
   return (
     <div style={{ width: "80%", margin: "25px auto auto auto" }}>
-      <Heading
-        borderBottom="1px"
-        borderColor="gray.200"
-        style={{ marginBottom: "25px" }}
-      >
+      <Heading borderBottom="1px" borderColor="gray.200" style={{ marginBottom: "25px" }}>
```
Comment (HaGuesto) [inline statviz/src/views/Dashboard/Dashboard.tsx:15]: I would have used here the divider from chakra

**Review by HaGuesto — COMMENTED** [review]
Interesting that there are so many auto removals of disable eslint lines

probably we should globalize generated types

Why did you not move your components to shared-components?

Error and loading states

globalize theming

try to not customize styling, check if you should adjust the theming.

**Comment by MaikNeubert** [comment]
Actions before merge: 

- [ ]  Deps nicht mehr in root package.json
- [ ]  backend deps where updated
- [ ]  fix front es-lint errors

File: statviz/src/views/Dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -44,10 +31,11 @@ export default function Dashboard() {
         </WrapItem>
```
Comment (HaGuesto) [inline statviz/src/views/Dashboard/Dashboard.tsx:31]: FYI: changes to this TimeRangeSelect will cause a re-render of the Dashboard component  and subsequently cause a query to be sent.

**Review by HaGuesto — COMMENTED** [review]
folder structure, check our FE README

generally try to use rem or the values from chakra ui instead of px, because they do not scale.

sometimes it is a bit confusing where data is processed

**Comment by MaikNeubert** [comment]
@HaGuesto 
the most importand issues are fixed and this PR can now be merged.
Note that this merge request is for merging into the shared-components-package-research branch.
You can either merge the shared-components-package-research first and change this PR to merge into master or first merge this PR as it is and them merge the shared-components-package-research branch into master

**Pull request merged**